src/core/ApolloClient.ts

```diff
@@ -439,9 +439,10 @@ export class ApolloClient<TCacheShape> implements DataProxy {
    */
   public writeFragment<TData = any, TVariables = OperationVariables>(
     options: DataProxy.WriteFragmentOptions<TData, TVariables>,
-  ): void {
-    this.cache.writeFragment<TData, TVariables>(options);
+  ): Reference | undefined {
+    const ref = this.cache.writeFragment<TData, TVariables>(options);
     this.queryManager.broadcastQueries();
```

Comment:
```suggestion
    if (options.broadcast !== false) {
      this.queryManager.broadcastQueries();
    }
```
While you're here, here's another alignment/consistency issue. Requesting `broadcast: false` should prevent this `broadcastQueries` call.

Reply:
Thanks, I just realised that `writeQuery` suffers the same set of inconsistencies, should update both and re-title the PR?

Reply:
@andrebrantom Yes please!